### ISSUE TITLE
New device support suggestion: Ailunce HD1

### ISSUE THREAD

**Issue by allesand** [hmatuschek/qdmr#411]
Since the radio is available and quite common, I am surprised that - from what I could search - it has not been requested before. So maybe this as a serach engine hint, wait and see if there are any future requests.

This is the one:
https://www.ailunce.com/Product/HD1/Overview
https://www.ailunce.com/Product/HD1/Specification
https://www.ailunce.com/Support/HD1/Software

**Comment by tomtom5152** [comment]
Would also be interested in seeing this as the Ailunce CPS is... not great...

Happy to help out with the coding if somebody can point me in the right direction for how to decode the `.tw` file. I suspect it is very similar to other Retevis files and the Ailunce website itself provides HD1 files on the HD2 codeplug page.

**Comment by hmatuschek** [comment]
Actually, the first thing I need would be wireshark capture of the USB traffic to the device. With this, I can better estimate the complexity of the implementation.

**Comment by fenfir** [comment]
There is this implementation that I've been trying to unravel. https://github.com/Avrahqedivra

I will try and get you a usb pcap this week.

**Comment by fenfir** [comment]
USB pcaps are here https://github.com/fenfir/pylunce/tree/main/dump

**Comment by fenfir** [comment]
I noticed after many frustrating attempts that there is some thing odd about the serial connection. It only works in windows. I get the correct version string in windows, but not on macos or linux, with the same python code.

**Comment by fenfir** [comment]
Looking into this a bit more, I found the problem with the serial connection. It uses a weird baud rate `117100` After that I noticed that you already have a c7k interface. HD2 is based on the c7k and the process there checks out with the other implementation I posted.

**Comment by hmatuschek** [comment]
Okay, thanks for diving into it. The protocol is indeed already implemented in the C7000Device class. However, it is tied to the GD73 interface. The best option would be, to separate the C7000 protocol from the GD73 interface. How does the HD2 register itself to the host? From your messages, I assume, that a CH380-like USB-to-Serial chip is used?

**Comment by fenfir** [comment]
Yes, it's a ch340 serial interface, then it does c7k protocol over serial at 117100 baud.

**Comment by fenfir** [comment]
There is also a 'GetVer' call that can be made outside of programming mode to confirm the running firmware.

**Comment by fenfir** [comment]
I have spent wayyy too much time in the last weeks working on documenting this codeplug. Here is a dump

https://fenfir.github.io/hd2_infodump/

**Comment by wykeradio** [comment]
> I have spent wayyy too much time in the last weeks working on documenting this codeplug. Here is a dump
> 
> https://fenfir.github.io/hd2_infodump/

Holy Cow.
Thats some serious debugging.

Thank you for your time and effort. It is very much appreciated by the community.
Regards, Tim
